hibernate-core/src/main/java/org/hibernate/loader/Loader.java

```diff
@@ -1666,50 +1663,46 @@ protected void instanceAlreadyLoaded(
 			);
 		}
 
-		if ( persister.getBytecodeEnhancementMetadata().isEnhancedForLazyLoading() && enhancementAsProxyEnabled ) {
-			if ( "merge".equals( session.getLoadQueryInfluencers().getInternalFetchProfile() ) ) {
-				assert this instanceof CascadeEntityLoader;
-				// we are processing a merge and have found an existing "managed copy" in the
-				// session - we need to check if this copy is an enhanced-proxy and, if so,
-				// perform the hydration just as if it were "not yet loaded"
-				final PersistentAttributeInterceptable interceptable = (PersistentAttributeInterceptable) object;
-				final PersistentAttributeInterceptor interceptor = interceptable.$$_hibernate_getInterceptor();
-				if ( interceptor instanceof EnhancementAsProxyLazinessInterceptor ) {
-					hydrateEntityState(
-							rs,
-							i,
-							persister,
-							getEntityAliases()[i].getRowIdAlias(),
-							key,
-							hydratedObjects,
-							session,
-							getInstanceClass(
-									rs,
-									i,
-									persister,
-									key.getIdentifier(),
-									session
-							),
-							object,
-							requestedLockMode
-					);
+		if (persister.getBytecodeEnhancementMetadata().isEnhancedForLazyLoading() && enhancementAsProxyEnabled) {
```

Comment:
It might be considerable approach to put `enhancementAsProxyEnabled` to left of `&&` for its performance cost is lower and we might end up with performance gain due to short-circuiting.

Reply:
I'll fix it thanks